benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksTask.java

```diff
@@ -0,0 +1,207 @@
+            // no-op
+          })
+          .subscribe();
+
+      if (executionTaskInterval == null || executionTaskInterval.isZero()) {
+        scheduler.schedule(this);
+      } else {
+        scheduler.schedule(this, executionTaskInterval.toMillis(), TimeUnit.MILLISECONDS);
+      }
+
+      return;
+    }
+
+    if (setStarted()) { // started
+      scheduledCompletingTask
+          .set(scheduler.schedule(this::startCompleting, executionTaskDuration.toMillis(), TimeUnit.MILLISECONDS));
+
+      try {
+        Mono<T> supplierMono = supplier.apply(benchmarksState);
+        if (supplierMono == null) {
+          startCompleting();
+          return;
+        }
+        supplierMono
+            .doOnError(this::startCompletingWithError)
+            .subscribe(res -> {
+              if (res == null) { // wtf
+                startCompleting();
+              } else if (setScheduled()) { // scheduled
+                supplierResult.set(res);
+                scheduler.schedule(this);
+              }
+            });
+      } catch (Throwable ex) {
+        startCompletingWithError(ex);
+      }
+    }
+  }
+
+  public Mono<Void> completionMono() {
+    return Mono.fromFuture(taskCompletionFuture);
+  }
+
+  private boolean setCompleted() {
+    boolean compareAndSet = taskStatus.compareAndSet(COMPLETING, COMPLETED);
+    Disposable disposable = scheduledCompletingTask.get();
+    if (disposable != null) {
+      disposable.dispose();
+    }
+    taskCompletionFuture.obtrudeValue(null);
+    return compareAndSet;
+  }
+
+  private boolean setCompletedWithError(Throwable throwable) {
+    boolean compareAndSet = taskStatus.compareAndSet(COMPLETING, COMPLETED);
+    Disposable disposable = scheduledCompletingTask.get();
+    if (disposable != null) {
+      disposable.dispose();
+    }
+    taskCompletionFuture.obtrudeException(throwable);
+    return compareAndSet;
+  }
+
+  private boolean setStarted() {
+    return taskStatus.compareAndSet(null, STARTED);
+  }
+
+  private boolean setScheduled() {
+    return taskStatus.compareAndSet(STARTED, SCHEDULED);
+  }
+
+  private boolean trySetCompleting() {
+    return taskStatus.compareAndSet(null, COMPLETING)
+        || taskStatus.compareAndSet(STARTED, COMPLETING)
+        || taskStatus.compareAndSet(SCHEDULED, COMPLETING);
+  }
+
+  private boolean isCompleted() {
+    return taskStatus.get() == COMPLETED;
+  }
+
+  private boolean isScheduled() {
+    return taskStatus.get() == SCHEDULED;
+  }
+
+  private void startCompletingWithError(Throwable throwable) {
+    if (trySetCompleting()) {
+      T res = supplierResult.get();
+      if (res == null) {
+        setCompletedWithError(throwable);
+        return;
+      }
+      try {
+        Mono<Void> voidMono = cleanUp.apply(res);
+        if (voidMono == null) {
+          setCompletedWithError(throwable);
+          return;
+        }
+        voidMono.subscribe(
+            aVoid -> setCompletedWithError(throwable),
```

Comment:
it never happens, there is no way to create an instance of `Void.class` and the reactor doesn't support the `null` value.